Loop/MainThreadLoopBox.swift

```diff
@@ -0,0 +1,85 @@
+import ReactiveSwift
+import Foundation
+#if canImport(Combine)
+import Combine
+#endif
+
+@available(macOS 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+class MainThreadLoopBox<State, Event>: LoopBoxBase<State, Event> {
+    override var _mainThreadView: LoopBoxBase<State, Event> {
+        self
+    }
+
+    override var lifetime: Lifetime {
+        _lifetime
+    }
+
+    /// Loop Internal SPI
+    override var _current: State {
+        cached
+    }
+
+    #if canImport(Combine)
+    override var objectWillChange: ObservableObjectPublisher { willChange }
+    private var willChange = ObservableObjectPublisher()
+    #endif
+
+    override var producer: SignalProducer<State, Never> {
+        return SignalProducer { observer, lifetime in
+            precondition(Thread.isMainThread)
```

Comment:
Does it mean that one should always call `observe(on: UIScheduler())`?

Reply:
It is meant for SwiftUI bindings, which are always called on the main thread.